src/java.base/share/classes/java/io/FileDescriptor.java

```diff
@@ -59,15 +61,35 @@ public final class FileDescriptor {
     private List<Closeable> otherParents;
     private boolean closed;
 
+    /**
+     * Called by FileDispatcherImpl when the file descriptor is about to be closed natively.
+     */
+    public void markClosedByNIO() {
+        resource.closedByNIO = true;
+    }
+
     class Resource implements jdk.internal.crac.JDKResource {
+        private static final boolean COLLECT_FD_STACKTRACES =
+                GetBooleanAction.privilegedGetProperty(JDKContext.COLLECT_FD_STACKTRACES_PROPERTY);
```

Comment:
Should not be moved to JDKContext or jdk.internal.Core ? To avoid any possible user to get the property value the similar way

Reply:
Not sure what you're trying to prevent, user code reading a property? The field here is private.

Reply:
My note is minor, I'm trying to save future users of COLLECT_FD_STACKTRACES_PROPERTY from readingthe property, instead the property can be used, parsed in JDKContext 
```
class JDKContext {
        static final boolean COLLECT_FD_STACKTRACES =
               GetBooleanAction.privilegedGetProperty(JDKContext.COLLECT_FD_STACKTRACES_PROPERTY);
}

// In this class: just use JDKContext.COLLECT_FD_STACKTRACES without GetProperty
```